client/src/index.jsx

```diff
@@ -33,35 +33,37 @@ export const Holder = styled.div`
 export default class App extends Component {
   constructor( props ) {
     super( props );
-    const id = this.getPathname();
     this.state = {
-      pathname: id,
     };
   }
 
   componentDidMount() {
-    this.getPathname();
-    this.getListingData( this.state.pathname );
+    this.getPathname().then( () => {
+      this.getListingData();
+    } );
   }
 
-  getPathname = () => {
-    const id = window.location.pathname;
+  getPathname = () => new Promise( ( resolve ) => {
+    let id = window.location.pathname;
     if ( id === '/' ) {
-      return 1;
+      id = 1;
+    } else {
+      id = id.replace( /\//g, '' );
     }
-    return id.substr( 1 );
-  }
+    resolve( this.setState( {
+      pathname: id,
+    } ) );
+  } )
 
-  getListingData = ( id ) => {
-    let idToUse = id;
-    if ( !idToUse ) {
-      idToUse = 1;
-    }
-    axios.get( `http://127.0.0.1:3002/rooms/${ idToUse }` )
+  getListingData = () => {
+    axios.get( `http://127.0.0.1:3002/rooms/${ this.state.pathname }/bookingInfo/` )
```

Comment:
I think you can use a relative path here (ie you don't need the http://127.0.0.1:3002 part) (?)

Reply:
You're right-- this would be better!